desktop section revamp (#39)
cryptomator/docs: source/desktop/volume-type.rst

.. _desktop/volume-type/general-volume-type-selection:

Volume Types
============

Volume types play an important role when handling your files.

When you unlock a vault, Cryptomator makes decrypted files available in your file manager by mounting a virtual drive on your operating system.

> This mounting of a virtual drive is handled differently depending on the volume type chosen in Cryptomator's settings.

Reviewer: ```suggestion
This mounting of a virtual drive is handled differently depending on the volume type chosen in Cryptomator's preferences.
```

WebDAV has widespread support and adequate performance, but its implementation differs between operating systems.

.. _desktop/volume-type/fuse:

What is a FUSE volume type?
---------------------------

Filesystem in Userspace (`FUSE <https://en.wikipedia.org/wiki/Filesystem_in_Userspace>`_) is a filesystem interface originally developed for Unix operating systems that lets non-privileged users create their own file systems without editing kernel code.
Which means, FUSE does not require admin privileges and is has good support across all major desktop operating systems.
FUSE volume type also delivers good performance when working on files.

All FUSE related volume types support custom mount options, but every option must be prefixed with ``-o``.
For example, you must enter ``-oallow_other`` if you want to specify ``allow_other`` option.

Choosing a default Volume Type
------------------------------

Cryptomator allows you to choose a default common volume type for all vaults.
In Cryptomator's window, navigate to ``Preferences`` (gear icon at top right), then ``Virtual Drive`` to set a default volume type.

> The availability of volume types depends on your operating system and installed libraries.

Reviewer: ```suggestion
The availability of volume types depends on your operating system and installed drivers.
```

Volume Type
===========

> When a vault is unlocked it is integrated (also called mounted) into the operating system as a virtual drive, so you can browse the decrypted files.

Reviewer: ```suggestion
When a vault is unlocked, it is integrated (also called mounted) into the operating system as a virtual drive, so you can browse the decrypted files.
```

Cryptomator currently offers three *volume types* used for mounting a vault. 
Each volume type has its own requirements, settings, benefits, and drawbacks.

#. :ref:`WebDAV <desktop/volume-type/webdav>`
#. :ref:`FUSE <desktop/volume-type/fuse>`
#. :ref:`Dokany <desktop/volume-type/dokany>`

.. _desktop/volume-type/general-volume-type-selection:

> Default Volume Type

Reviewer 2: This section is not about the "Default" volume type, but about general information about volume types

-------------------

Cryptomator allows you to select a default volume type to be used every time you unlock a vault.
You can set a default volume type by navigating to ``Preferences`` (gear icon at top right), then ``Virtual Drive``.

> The availabe volume types depend on your operating system and if required libraries are installed (e.g., macFUSE-T).

Reviewer: ```suggestion
The availabe volume types depend on your operating system and if required libraries are installed (e.g., FUSE-T).
```

Some volume type changes require an app restart, and you will be notified if a restart is required.

.. image:: ../img/desktop/preferences-virtual-drive.png
    :alt: Virtual Drive Tab in Preferences

Volume types, like WebDAV, etc, allow specifying custom mount options.
If supported, these options can be set for each vault individually in :ref:`desktop/vault-management/vault-options`, under the mounting tab.

.. _desktop/volume-type/webdav:

> WebDAV Volume Type

Reviewer 2: i suggest to use plural: "WebDAV Volume Types"

Reviewer 3: I plead to remove this subsection altogether.

Instead of 

```txt
├─ volume types
│  ├─ general
│  ├─ fuse
│  │  ├─ fuse (linux)
│  │  ├─ fuse-t (mac)
│  │  ├─ macfuse (mac)
│  │  ├─ winfsp (win)
│  ├─ webdav
│  │  ├─ webdav (windows)
│  │  ├─ webdav (gio)
│  │  ├─ webdav (apple script)
│  │  ├─ webdav (http address)
```

I would rather see:

```txt
├─ general
├─ volume types
│  ├─ fuse (linux)
│  ├─ fuse-t (mac)
│  ├─ macfuse (mac)
│  ├─ winfsp (win)
│  ├─ webdav (windows)
│  ├─ webdav (gio)
│  ├─ webdav (apple script)
│  ├─ webdav (http address)
```

Reviewer 3: > What's the problem with current structure?

The documentation should reflect the structure in the software. And the software doesn't have such a sub-hierarchy. All implementations are 100% exchangeable. From a user's perspective it doesn't matter what kind of protocol is used internally. This would be like grouping applications by the programming language used, while the user really just wants to know whether the application is a game or a text editor.

I don't see any benefit in explaining what FUSE and WebDAV is _in general_. Instead we should focus on what are the pros and cons of each _individual implementation_ are.

Reviewer 3: In that case, how about putting the "general" info about FUSE and WebDAV into the "general" section as well? It is just one sentence each, anyway.

Reviewer 3: Sure, go ahead 🙂

WebDAV has widespread support and adequate performance, but its implementation differs between operating systems.

.. _desktop/volume-type/webdav/explorer:

WebDAV (Windows Explorer)
^^^^^^^^^^^^^^^^^^^^^^^^^

**Requirements:** Windows

WebDAV on Windows uses the `net use <https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2012-R2-and-2012/gg651155(v=ws.11)>`_ command to handle WebDAV operations.
By default, unlocked vaults are mounted as a network drive and assigned a random drive letter.
Using WebDAV on Windows has the following drawbacks:

    * The size of transferred files is restricted to a maximum of 4GB.
    * The total space and free space of the network drive are shown to be the same as the total space and free space of the C: drive, which is technically incorrect.

.. Additionally, sometimes mounting fails with ``System error 67 has occurred. The network name cannot be found.`` (or its translations).
.. If it happens, follow you can follow the guide TODO to get access again.

.. _desktop/volume-type/webdav/applescript:

WebDAV (AppleScript)
^^^^^^^^^^^^^^^^^^^^

**Requirements:** macOS

> WebDAV on utlizes the scripting language `AppleScript` to handle WebDAV operations.

Reviewer: ```suggestion
WebDAV on macOS utlizes the scripting language `AppleScript` to mount/unmount the virtual drive.
```

**Requirements:** Linux, ``gio`` installed

Due to the wide variety of Linux distributions, Cryptomator only supports system integrated WebDAV volume type if `gio <https://manpage.me/?gio>`_ is installed.
You can unlock your vault without ``gio`` using :ref:`desktop/volume-type/webdav/http-address`, but support across distributions is not gurranteed.
Also, it's upto yourself to figure out how to integrate WebDAV share with your distro.

.. _desktop/volume-type/webdav/http-address:

WebDAV (HTTP Address)
^^^^^^^^^^^^^^^^^^^^^

**Requirements:** None

This volume type is always present and comes in handy when all other volume types fails to mount.
It starts the above mentioned local-only WebDAV server, which can be manually integrated into the system or accessed using a third party application, like `CyberDuck <https://cyberduck.io/>`_).

.. _desktop/volume-type/fuse:

> FUSE Volume Type

Reviewer 2: i suggest to use plural: "FUSE Volume Types"

Reviewer 3: (see https://github.com/cryptomator/docs/pull/39#discussion_r1193682340)

**Requirements:** MacOS, FUSE-T installed

This volume type depends on a library provided by the new `FUSE-T project <https://www.fuse-t.org/>`_.
You can install it using brew:

.. code-block:: shell

    brew tap macos-fuse-t/homebrew-cask
    brew install fuse-t

By default, unlocked vaults are mounted to `/Volumes`.
Info on custom mount options is available at `wiki of the osxfuse project <https://github.com/osxfuse/osxfuse/wiki/Mount-options>`_.

.. note::

>     FUSE-T is a new project, so you may come across bugs here and there.

Reviewer 2: "here and there" is a quite... lightweight formulation. Since users entrust Cryptomator with their data, i suggest to be more serious.

**Requirements:** Windows, WinFsp installed

> FUSE with WinFsp depends on a library provided by the `WinFsp project <https://winfsp.dev/>`_.

Reviewer 2: ```diff
+ This volume type binds to a library provided by the `WinFsp project <https://winfsp.dev/>`_.
-  The `WinFsp project <https://winfsp.dev/>`_ provides FUSE bindings for Windows.
```

.. warning::

>     Dokany support in Cryptomator is deprecated since version 1.7.0. We suggest a to switch to :ref:`desktop/volume-type/fuse/winfsp`.

Reviewer 2: ```diff
-     Dokany support in Cryptomator is deprecated since version 1.7.0. We suggest a to switch to :ref:`desktop/volume-type/fuse/winfsp`.
+     Dokany support in Cryptomator is deprecated since version 1.7.0. We suggest to switch to :ref:`desktop/volume-type/fuse/winfsp`.
